make compile fails on Linux

make uses PRESET Debug, where as the CI uses Release.

**Debug**
```
In file included from /usr/include/c++/13/cassert:44,
                 from /home/personal/repos/sourcemeta/blaze/src/compiler/compile_describe.cc:4:
/home/personal/repos/sourcemeta/blaze/src/compiler/compile_describe.cc: In function ‘std::string sourcemeta::blaze::describe(bool, const Instruction&, const sourcemeta::core::WeakPointer&, const sourcemeta::core::WeakPointer&, const sourcemeta::core::JSON&, const sourcemeta::core::JSON&)’:
/home/personal/repos/sourcemeta/blaze/src/compiler/compile_describe.cc:1342:22: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
 1342 |       assert(maximum >= 0);
      |              ~~~~~~~~^~~~
/home/personal/repos/sourcemeta/blaze/src/compiler/compile_describe.cc:1374:20: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
 1374 |     assert(minimum >= 0);
      |            ~~~~~~~~^~~~
/home/personal/repos/sourcemeta/blaze/src/compiler/compile_describe.cc:1409:22: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
 1409 |       assert(maximum >= 0);
      |              ~~~~~~~~^~~~
/home/personal/repos/sourcemeta/blaze/src/compiler/compile_describe.cc:1461:22: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
 1461 |       assert(minimum >= 0);
      |              ~~~~~~~~^~~~
[ 66%] Building CXX object src/compiler/CMakeFiles/sourcemeta_blaze_compiler.dir/compile_output_trace.cc.o
cc1plus: all warnings being treated as errors
gmake[3]: *** [src/compiler/CMakeFiles/sourcemeta_blaze_compiler.dir/build.make:90: src/compiler/CMakeFiles/sourcemeta_blaze_compiler.dir/compile_describe.cc.o] Error 1
gmake[3]: *** Waiting for unfinished jobs....
gmake[3]: Leaving directory '/home/personal/repos/sourcemeta/blaze/build'
gmake[2]: *** [CMakeFiles/Makefile2:1007: src/compiler/CMakeFiles/sourcemeta_blaze_compiler.dir/all] Error 2
gmake[2]: *** Waiting for unfinished jobs....
[ 67%] Linking CXX static library libsourcemeta_core_alterschema.a
gmake[3]: Leaving directory '/home/personal/repos/sourcemeta/blaze/build'
[ 67%] Built target sourcemeta_core_alterschema
gmake[2]: Leaving directory '/home/personal/repos/sourcemeta/blaze/build'
gmake[1]: *** [Makefile:146: all] Error 2
gmake[1]: Leaving directory '/home/personal/repos/sourcemeta/blaze/build'
make: *** [Makefile:23: compile] Error 2
```


Attaching the make configure output for reference
```
make configure
cmake -S . -B ./build \
        -DCMAKE_BUILD_TYPE:STRING=Debug \
        -DCMAKE_COMPILE_WARNING_AS_ERROR:BOOL=ON \
        -DBLAZE_TESTS:BOOL=ON \
        -DBLAZE_BENCHMARK:BOOL=ON \
        -DBLAZE_CONTRIB:BOOL=ON \
        -DBLAZE_DOCS:BOOL=ON \
        -DBUILD_SHARED_LIBS:BOOL=OFF
-- The CXX compiler identification is GNU 13.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The C compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Failed to find LLVM FileCheck
-- Found Git: /usr/bin/git (found version "2.43.0") 
-- Google Benchmark version: v1.8.5, normalized to 1.8.5
-- Looking for shm_open in rt
-- Looking for shm_open in rt - found
-- Performing Test HAVE_CXX_FLAG_WALL
-- Performing Test HAVE_CXX_FLAG_WALL - Success
-- Performing Test HAVE_CXX_FLAG_WEXTRA
-- Performing Test HAVE_CXX_FLAG_WEXTRA - Success
-- Performing Test HAVE_CXX_FLAG_WSHADOW
-- Performing Test HAVE_CXX_FLAG_WSHADOW - Success
-- Performing Test HAVE_CXX_FLAG_WFLOAT_EQUAL
-- Performing Test HAVE_CXX_FLAG_WFLOAT_EQUAL - Success
-- Performing Test HAVE_CXX_FLAG_WOLD_STYLE_CAST
-- Performing Test HAVE_CXX_FLAG_WOLD_STYLE_CAST - Success
-- Performing Test HAVE_CXX_FLAG_WCONVERSION
-- Performing Test HAVE_CXX_FLAG_WCONVERSION - Success
-- Performing Test HAVE_CXX_FLAG_WERROR
-- Performing Test HAVE_CXX_FLAG_WERROR - Success
-- Performing Test HAVE_CXX_FLAG_WSUGGEST_OVERRIDE
-- Performing Test HAVE_CXX_FLAG_WSUGGEST_OVERRIDE - Success
-- Performing Test HAVE_CXX_FLAG_PEDANTIC
-- Performing Test HAVE_CXX_FLAG_PEDANTIC - Success
-- Performing Test HAVE_CXX_FLAG_PEDANTIC_ERRORS
-- Performing Test HAVE_CXX_FLAG_PEDANTIC_ERRORS - Success
-- Performing Test HAVE_CXX_FLAG_WSHORTEN_64_TO_32
-- Performing Test HAVE_CXX_FLAG_WSHORTEN_64_TO_32 - Failed
-- Performing Test HAVE_CXX_FLAG_FSTRICT_ALIASING
-- Performing Test HAVE_CXX_FLAG_FSTRICT_ALIASING - Success
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS - Success
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED - Success
-- Performing Test HAVE_CXX_FLAG_WSTRICT_ALIASING
-- Performing Test HAVE_CXX_FLAG_WSTRICT_ALIASING - Success
-- Performing Test HAVE_CXX_FLAG_WD654
-- Performing Test HAVE_CXX_FLAG_WD654 - Failed
-- Performing Test HAVE_CXX_FLAG_WTHREAD_SAFETY
-- Performing Test HAVE_CXX_FLAG_WTHREAD_SAFETY - Failed
-- Performing Test HAVE_CXX_FLAG_COVERAGE
-- Performing Test HAVE_CXX_FLAG_COVERAGE - Success
-- Compiling and running to test HAVE_STD_REGEX
m-- Performing Test HAVE_STD_REGEX -- success
-- Compiling and running to test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile
-- Compiling and running to test HAVE_POSIX_REGEX
a-- Performing Test HAVE_POSIX_REGEX -- success
-- Compiling and running to test HAVE_STEADY_CLOCK
k-- Performing Test HAVE_STEADY_CLOCK -- success
-- Compiling and running to test HAVE_PTHREAD_AFFINITY
e-- Performing Test HAVE_PTHREAD_AFFINITY -- success
-- Performing Test COMPILER_SUPPORTS_AVX2
 -- Performing Test COMPILER_SUPPORTS_AVX2 - Success
-- Enabling SIMD AVX2
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Configuring done (3.3s)
-- Generating done (0.0s)
-- Build files have been written to: /home/personal/repos/sourcemeta/blaze/build
```

--- a/src/compiler/compile_describe.cc
+++ b/src/compiler/compile_describe.cc
@@ -1339,7 +1339,6 @@ auto describe(const bool valid, const Instruction &step,
       std::ostringstream message;
       const auto maximum{instruction_value<ValueUnsignedInteger>(step) - 1};
       message << "The array value was expected to contain at most " << maximum;
-      assert(maximum >= 0);
       if (maximum == 1) {
         message << " item";
       } else {
@@ -1371,7 +1370,6 @@ auto describe(const bool valid, const Instruction &step,
     std::ostringstream message;
     const auto minimum{instruction_value<ValueUnsignedInteger>(step) + 1};
     message << "The array value was expected to contain at least " << minimum;
-    assert(minimum >= 0);
     if (minimum == 1) {
       message << " item";
     } else {
@@ -1406,7 +1404,6 @@ auto describe(const bool valid, const Instruction &step,
       std::ostringstream message;
       const auto maximum{instruction_value<ValueUnsignedInteger>(step) - 1};
       message << "The object value was expected to contain at most " << maximum;
-      assert(maximum >= 0);
       if (maximum == 1) {
         message << " property";
       } else {
@@ -1458,7 +1455,6 @@ auto describe(const bool valid, const Instruction &step,
       const auto minimum{instruction_value<ValueUnsignedInteger>(step) + 1};
       message << "The object value was expected to contain at least "
               << minimum;
-      assert(minimum >= 0);
       if (minimum == 1) {
         message << " property";
       } else {

